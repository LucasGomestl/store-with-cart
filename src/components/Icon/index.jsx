import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Link } from "react-router-dom";

const Icon = ({ icon, link, color }) => {
  return (
    <>
      {link ? (
        <Link to={link} data-testid="link">
          <FontAwesomeIcon icon={icon} color={color}></FontAwesomeIcon>
        </Link>
      ) : (
        <FontAwesomeIcon icon={icon} color={color}></FontAwesomeIcon>
      )}
    </>
  );
};

export default Icon;
