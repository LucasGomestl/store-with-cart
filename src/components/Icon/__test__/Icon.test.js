import React from "react";
import { render } from "@testing-library/react";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import Icon from "../index";

describe("Icon tests", () => {
  it("Should render the Icon", () => {
    const { container, queryByTestId } = render(<Icon icon={faSearch} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(queryByTestId("link")).not.toBeInTheDocument();
  });

  it("Should render the Icon with a link around", () => {
    const { getByTestId } = render(<Icon icon={faSearch} link="/" />);
    expect(getByTestId("link")).toBeInTheDocument();
  });
});
