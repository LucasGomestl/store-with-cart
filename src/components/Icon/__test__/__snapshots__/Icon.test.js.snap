// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon tests Should render the Icon 1`] = `
<i
  class="fa"
/>
`;
